src/modules/Update.rb

```diff
@@ -222,24 +223,14 @@ def IsProductSupportedForUpgrade
     #-----------------------------------------------------------------------
 
     def SelectedProducts
-      selected = Pkg.ResolvableProperties("", :product, "")
-      selected = Builtins.filter(selected) do |p|
-        Ops.get(p, "status") == :selected
-      end
-      Builtins.maplist(selected) do |p|
-        Ops.get_locale(
-          p,
-          "display_name",
-          Ops.get_locale(
-            p,
-            "summary",
-            Ops.get_locale(
-              p,
-              "name",
-              Ops.get_locale(p, "version", _("Unknown Product"))
-            )
-          )
-        )
+      selected = Y2Packager::Resolvable.find(kind: :product, status: :selected)
+
+      selected.map do |p|
```

Comment:
np: what about using `product` instead of just `p`? Please, feel free to ignore this suggestion :)